File: src/harvest/promote/yaml-scalar.ts
/** One frontmatter scalar. Strings are always double-quoted with JSON escaping,
 * which is valid YAML and sidesteps every case where an unquoted value would be
 * reinterpreted - a title starting with `-`, one reading `yes` or `null`, one
 * containing a colon. Medium titles hit all three. */
export const yamlScalar = (value: string | number | boolean | null): string => {
  if (value === null) return 'null'
  if (typeof value === 'string') return JSON.stringify(value)
  return String(value)
}

File: tools/capture/yaml_scalar.py
"""Yaml scalar: extracted from normalize_thin_clip.py."""

import json


def yaml_scalar(value: object) -> str:
    """One frontmatter value the way `buildFrontmatter` writes it: null, bool, JSON string, or bare."""
    if value is None:
        return "null"
    if isinstance(value, bool):
        return "true" if value else "false"
    if isinstance(value, str):
        return json.dumps(value)
    return str(value)
